dcm/src/main/java/org/dcm/viewupdater/DDlogUpdater.java

```diff
@@ -0,0 +1,121 @@
+        try {
+            API = new DDlogAPI(ddlogWorkerThreads, null, storeDataInDDlogBackgroundProgram);
+        } catch (final DDlogException e) {
+            throw new RuntimeException(e);
+        }
+
+        this.consumer = consumer;
+        this.tableIDMap = new HashMap<>();
+        this.irTables = irTables;
+    }
+
+    private DDlogRecord toDDlogRecord(final String tableName, final Object[] args) {
+        final DDlogRecord[] recordsArray = new DDlogRecord[args.length];
+        final IRTable irTable = irTables.get(tableName);
+        final Table<? extends Record> table = irTable.getTable();
+
+        int counter = 0;
+        for (final Field<?> field : table.fields()) {
+            final Class<?> cls = field.getType();
+            try {
+                switch (cls.getName()) {
+                    case BOOLEAN_TYPE:
+                        recordsArray[counter] = new DDlogRecord((Boolean) args[counter]);
+                        break;
+                    case INTEGER_TYPE:
+                        recordsArray[counter] = new DDlogRecord((Integer) args[counter]);
+                        break;
+                    case LONG_TYPE:
+                        recordsArray[counter] = new DDlogRecord((Long) args[counter]);
+                        break;
+                    case STRING_TYPE:
+                        recordsArray[counter] = new DDlogRecord(args[counter].toString().trim());
+                        break;
+                    default:
+                        throw new RuntimeException("Unexpected datatype: " + cls.getName());
+                }
+            } catch (final DDlogException e) {
+                throw new RuntimeException(e);
+            }
+            counter = counter + 1;
+        }
+        DDlogRecord record = null;
+        try {
+            record = DDlogRecord.makeStruct(tableName, recordsArray);
+        } catch (final DDlogException e) {
+            throw new RuntimeException(e);
+        }
+        return record;
+    }
+
+    public void sendUpdatesToDDlog(final Map<String, List<Object[]>> commands) {
+        final int counter = commands.values().stream().mapToInt(List::size).sum();
+
+        final DDlogRecCommand[] ddlogCommands = new DDlogRecCommand[counter];
+        int loopCounter = 0;
+        for (final Map.Entry<String, List<Object[]>> entry : commands.entrySet()) {
+            final String tableName = entry.getKey();
+            final List<Object[]> cmds = entry.getValue();
+
+            int id;
+            if (!tableIDMap.containsKey(tableName)) {
+                id = API.getTableId(tableName);
+                tableIDMap.put(tableName, id);
+            }
+            id = tableIDMap.get(tableName);
+
+            for (int i = 0; i < cmds.size(); i++) {
+                ddlogCommands[loopCounter] =
+                        new DDlogRecCommand(DDlogCommand.Kind.Insert, id, toDDlogRecord(tableName, cmds.get(i)));
```

Comment:
This is a bug. Why are these always inserts? What about other deletions? Which basically means there is no test that is exercising deletions. :)